File: 🟢 Easy/349 - Intersection of Two Arrays/_349IntersectionofTwoArrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        num1 = set(nums1)
        num2 = set(nums2)
        res = []

        for n in num1:
            if n in num2:
                res.append(n)

        return res

File: 🟢 Easy/349 - Intersection of Two Arrays/_349IntersectionofTwoArrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> num1(nums1.begin(), nums1.end());
        unordered_set<int> num2(nums2.begin(), nums2.end());

        vector<int> res;
        for (auto& n : num1) {
            if (num2.find(n) != num2.end()) {
                res.push_back(n);
            }
        }

        return res;
    }
};